backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BinaryInterface.kt

```diff
@@ -112,14 +112,19 @@ internal val IrClass.kotlinObjCClassInfoSymbolName: String
         return "kobjcclassinfo:$fqNameForIrSerialization"
     }
 
-val IrFunction.functionName get() = with(KonanBinaryInterface) { functionName }
-val IrFunction.fullName get() = parent.fqNameForIrSerialization.child(Name.identifier(functionName)).asString()
+fun IrFunction.computeFunctionName() = with(KonanBinaryInterface) { functionName }
 
-val IrFunction.symbolName get() = with(KonanBinaryInterface) { symbolName }
+fun IrFunction.computeFullName() = parent.fqNameForIrSerialization.child(Name.identifier(computeFunctionName())).asString()
 
-val IrField.symbolName get() = with(KonanBinaryInterface) { symbolName }
+fun IrFunction.computeSymbolName() = with(KonanBinaryInterface) { symbolName }.replaceSpecialSymbols()
 
-val IrClass.typeInfoSymbolName get() = with(KonanBinaryInterface) { typeInfoSymbolName }
+fun IrField.computeSymbolName() = with(KonanBinaryInterface) { symbolName }.replaceSpecialSymbols()
+
+fun IrClass.computeTypeInfoSymbolName() = with(KonanBinaryInterface) { typeInfoSymbolName }.replaceSpecialSymbols()
+
+private fun String.replaceSpecialSymbols() =
+        // '@' is used for symbol versioning in GCC: https://gcc.gnu.org/wiki/SymbolVersioning.
+        this.replace("@", "__")
```

Comment:
So here we generally hope that this won't make symbol names clash?

Reply:
Yes. The tricky part here are functions that contain `@` in their name. Maybe make it more robust and readable, e.g. `__at__`?

Reply:
Well, maybe. On the other hand, functions that contain `@` in their name probably could clash with extension functions before this patch. So let's avoid complicated mangling for now. `__at__` or similar things should be just ok.